contracts/BSCValidatorSet.sol

```diff
@@ -450,15 +459,49 @@ contract BSCValidatorSet is IBSCValidatorSet, System, IParamSubscriber, IApplica
     return isWorkingValidator(index);
   }
 
+  function distributeFinalityReward(address[] calldata valAddrs, uint256[] calldata weights) external onlyCoinbase onlyInit {
+    uint256 totalValue;
+    totalValue = (address(SYSTEM_REWARD_ADDR).balance * finalityRewardRatio) / 100;
+    totalValue = ISystemReward(SYSTEM_REWARD_ADDR).claimRewards(payable(address(this)), totalValue);
+
+    if (totalValue > 0) {
+      uint256 value;
+      address valAddr;
+      uint256 index;
+      uint256 totalWeight;
+      for (uint256 i = 0; i < weights.length; i++) {
+        totalWeight += weights[i];
+      }
+      for (uint256 i = 0; i < valAddrs.length; i++) {
```

Comment:
Handle if `totalWeight` is zero to avoid dive 0

Reply:
Fixed